src/payment/strategies/klarnav2/klarnav2-payment-strategy.ts

```diff
@@ -30,11 +31,31 @@ export default class KlarnaV2PaymentStrategy implements PaymentStrategy {
     initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
         return this._klarnav2ScriptLoader.load()
             .then(klarnaPayments => { this._klarnaPayments = klarnaPayments; })
-            .then(() => this._loadPaymentsWidget(options))
+            .then(() => {
+                let currentBalance: number | undefined;
+
+                this._unsubscribe = this._store.subscribe(
+                    state => {
+                        const prevBalance = currentBalance;
+                        const checkout = state.checkout.getCheckout();
+                        currentBalance = checkout && checkout.outstandingBalance;
+
+                        if (prevBalance && prevBalance !== currentBalance) {
```

Comment:
instead of doing this, you can pass a second parameter to `subscribe()` that will only trigger the callback when the returned value changes.

```js
this._unsubscribe = this._store.subscribe(
// callback,
({ checkout }) => {
  const checkout = checkout.getCheckout();

  return checkout && checkout.outstandingBalance;
}